contracts/CourtRoom.sol

```diff
@@ -49,6 +56,22 @@ contract CaseContract is Owned {
 
 	}
 
+	function submitPromise(bytes32 _claimId,address beneficiary, uint256 blockNumber,
```

Comment:
Yes, but it it is not just payment . it is a promise to provide a service up to block X and it is part of the claim validity check where the contract take decision upon.